File: Backend/src/main/java/com/example/demo/Models/MediaModel.java
package com.example.demo.Models;

import com.example.demo.DemoApplication;
import com.fasterxml.jackson.core.JsonProcessingException;

public class MediaModel {

    long id;

    
    String path;
    String type;
    PostModel postModel;


    public long getId() {
        return this.id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public PostModel getPostModel() {
        return this.postModel;
    }

    public void setPostModel(PostModel postModel) {
        this.postModel = postModel;
    }

    public MediaModel() {

    }

    public MediaModel(String path, String type) {
        this.path = path;
        this.type = type;
    }



    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }


    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        try {
            return DemoApplication.objectMapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }
        return null;
    }
}

File: Frontend/lib/Model/MediaModel.dart
import 'PostModel.dart';

class MediaModel {
  String? id;
  String? path;
  String? type;
  PostModel? postModel;

  MediaModel({this.id, this.path, this.type, this.postModel});

  MediaModel.fromJson(Map<String, dynamic> json) {
    id = json["id"].toString();
    path = json["path"];
    type = json["type"];
    if (json["postModel"] != null) {
      postModel = PostModel.fromJson(json["postModel"]);
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['path'] = path;
    data['type'] = type;
    if (postModel != null) data['postModel'] = postModel!.toJson();
    return data;
  }

  get getId => id;

  set setId(id) => this.id = id;

  get getImage => path;

  set setImage(path) => this.path = path;

  get getType => type;

  set setType(type) => this.type = type;

  get getPostModel => postModel;

  set setPostModel(postModel) => this.postModel = postModel;
}
